src/Interpreters/SelectQueryOptions.h

```diff
@@ -31,6 +31,8 @@ struct SelectQueryOptions
     bool only_analyze = false;
     bool modify_inplace = false;
     bool remove_duplicates = false;
+    /// This is required for UNION to match headers correctly.
+    bool strict_order = false;
```

Comment:
Variable name is unclear. I'd suggest something like `reorder_columns_as_reqiured_header`.